modules/kubernetes/datadog/main.tf

```diff
@@ -14,15 +14,149 @@ terraform {
 
   required_providers {
     kubernetes = {
-      source  = "hashicorp/kubernetes"
-      version = "2.13.1"
-    }
-    helm = {
-      source  = "hashicorp/helm"
-      version = "2.6.0"
+      source  = "xenitab/git"
+      version = "0.0.1"
     }
   }
 }
+   
+resource "git_repository_file" "kustomization" {
+ path = "clsuters/we-dev-aks1/datadog.yaml"
+ content = <<EOF
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
+kind: Kustomization
+metadata:
+  name: datadog
+  namespace: flux-system
+spec:
+  interval: 5m
+  sourceRef:
+    kind: GitRepository
+    name: flux-system
+  path: ./platform/we-dev-aks1/datadog
+  prune: true
+  validation: client
+ EOF
```

Comment:
I don't like doing these as strings. Either HCL to YAML or using templates. This feels like things like number of spaces will cause issues.